File: src/Java/801-900/852.PeakIndexInMountainArray.java

import java.awt.List;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.Collections;
import java.util.Comparator;
import java.util.LinkedList;
import java.util.Queue;
import java.util.Set;
import java.util.HashMap;
import java.util.HashSet;

class Solution {
    public int peakIndexInMountainArray(int[] A) {
        int left = 0;
        int right = A.length;
        while (left < right){
            int mid = (left + right) / 2;
            if (A[mid] < A[mid + 1])  
                left = mid + 1;
            else
                right = mid;
        }
        return left;
    }
}

File: src/Python/801-900/852.PeakIndexInMountainArray.py
class Solution:
    def peakIndexInMountainArray(self, A):
        """
        :type A: List[int]
        :rtype: int
        """
        left, right = 0, len(A)
        while left < right:
            mid = int((left + right) / 2)
            if A[mid] < A[mid + 1]:  
                left = mid + 1
            else:
                right = mid 
        return left


if __name__ == '__main__':
    solution = Solution()
    print(solution.peakIndexInMountainArray([0,2,1,0]))
else:
    pass
